### PR TITLE
Add a document exploring the risk of priority inversions

### PR DESCRIPTION
**Pull request by shaseley**
@jyasskin can you PTAL? Thank you!

### CODE REVIEW

File: misc/priority-inversion.md [hunk 1]
```diff
@@ -1 +1,251 @@
...
+
+...
+
+// A subsequent request for |url| happens just after, at user-blocking priority
+// This can lead to a priority inversion if the system is not designed to
+// handle this case.
+getCachedResource(url, 'user-blocking').then(...);
+```
+
+Developers should be aware that such situations could arise. Systems like this
```
Comment (natechapin) [inline misc/priority-inversion.md:154]: remove "like this"? It's probably clearer without.

File: misc/priority-inversion.md [hunk 2]
```diff
@@ -1 +1,251 @@
...
+that use internal priorities or a combination of `setTimeout()` and
+`requestIdleCallback()` might have already faced such issues. For
+`scheduler.postTask()`, there are a couple ways to mitigate this type of
+problem at the application-level:
+
+ * Detect that the situation and use the `TaskController`/`TaskSignal` API to
+   change the priority of underlying task(s). `scheduler.postTask()` was
+   designed around task priorities needing to be dynamic, such as in this
```
Comment (natechapin) [inline misc/priority-inversion.md:162]: ..."designed with dynamic task priorities for scenarios such as this one." ?

**Review by natechapin — APPROVED** [review]

File: explainers/prioritized-post-task.md [hunk 3]
```diff
@@ -503,19 +503,25 @@ lightweight solution and will publish an explainer soon.
 ### Is there a risk of priority inversions?
 
 Priority inversions occur whenever higher priority work depends on lower
-priority work, so simply by introducing priorities introduces this risk. There
-is only so much the API can do to prevent these situations, although [inheriting
-priority](post-task-propagation.md) can help to some degree.
+priority work, so simply by introducing priorities introduces this risk. But
+the lack of preemption on the web creates a different a fundamentally different
```
[the remark this answers is not in the document]
Comment (shaseley) [reply explainers/prioritized-post-task.md]: Done.

File: misc/priority-inversion.md (hunk 1 above)
Comment (shaseley) [reply misc/priority-inversion.md]: sgtm, done.

File: misc/priority-inversion.md (hunk 2 above)
Comment (shaseley) [reply misc/priority-inversion.md]: Nice; done.

**Comment by shaseley** [comment]
Thanks for the review, @natechapin!

**Comment by jyasskin** [comment]
Looks good!

**Pull request merged**